File: Leetcode/118 - Pascal's Triangle/solution.ts
function generate(numRows: number): number[][] {
    const pascalTriangle: number[][] = new Array(numRows);
    for (let i = 0; i < numRows; i++) {
        pascalTriangle[i] = new Array(i + 1).fill(1);
        for (let j = 1; j < i; j++) {
            pascalTriangle[i][j] = pascalTriangle[i - 1][j - 1] + pascalTriangle[i - 1][j];
        }
    }
    return pascalTriangle;
};

File: Leetcode/118 - Pascal's Triangle/solution.py
def generate(self, numRows: int) -> List[List[int]]:
    pascal_triangle: List[List[int]] = [[1] * (i + 1) for i in range(numRows)]
    for i in range(2, numRows):
        for j in range(1, i):
            pascal_triangle[i][j] = pascal_triangle[i - 1][j - 1] + pascal_triangle[i - 1][j]
    return pascal_triangle